packages/material-ui/src/ExpansionPanel/ExpansionPanel.js

```diff
@@ -96,6 +96,24 @@ const ExpansionPanel = React.forwardRef(function ExpansionPanel(props, ref) {
   const [expandedState, setExpandedState] = React.useState(defaultExpanded);
   const expanded = isControlled ? expandedProp : expandedState;
 
+  React.useEffect(() => {
+    if (process.env.NODE_ENV !== 'production') {
```

Comment:
This check should be around the `useEffect` call, see https://github.com/mui-org/material-ui/pull/17420

Reply:
I think that we should wait for the direction taken in #17420 to be approved before considering it here. I might have approved it too early 👼. I would be eager to have more than 2 people (you and me) weighing the pros and cons of the change. I don't remember why we decided this approach in the first place. I would hope we make it a mindful choice.